### PR TITLE
One namespace for component's Pipeline-as-Code CI

### PR DESCRIPTION
**Pull request by Michkov**
Unify .tekton directories into tekton-ci component.

Reasons:
- Shared-resources should not be used anymore, which would require to
  create/manage secret in each namespace. With one namespaces it would
  be managed only there
- SRE engineers are interested in the service, so they will see only the
  service in the service namespace - not pods of CI pipelines
- Quota can be set on service namespace, CI pipelineRuns will not take
  resources from it
- In case of multiple environments we can select one when the CI will be
  installed, as part of the commit the CI is disabled for developemnt
  environment

### CODE REVIEW

File: components/tekton-ci/repository.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  name: service-provider-integration
+spec:
+  url: "https://github.com/redhat-appstudio/service-provider-integration-operator"
+---
+apiVersion: pipelinesascode.tekton.dev/v1alpha1
+kind: Repository
+metadata:
+  name: e2e-tests
+spec:
+  url: "https://github.com/redhat-appstudio/e2e-tests"
```
Comment (bnallapeta) [inline components/tekton-ci/repository.yaml:21]: This is included by mistake? What would e2e folder contain after this change? Is it needed? (maybe in e2e repository?)
If not, need to remove this block as well as the e2e folder under components.

**Comment by scoheb** [comment]
what are the issues you are trying to solve with this or what are the benefits?

**Review by bnallapeta — COMMENTED** [review]
need to update e2e-tests to tekton-ci [here](https://github.com/Michkov/infra-deployments/blob/merged-tekton-ci/hack/destroy-cluster.sh#L44).

File: components/tekton-ci/repository.yaml (hunk 1 above)
Comment (Michkov) [reply components/tekton-ci/repository.yaml:21]: folder in components removed, but this code is needed, to be able to run pipelines defined in https://github.com/redhat-appstudio/e2e-tests/tree/main/.tekton

**Comment by Michkov** [comment]
> need to update e2e-tests to tekton-ci [here](https://github.com/Michkov/infra-deployments/blob/merged-tekton-ci/hack/destroy-cluster.sh#L44).

Updated

**Comment by Michkov** [comment]
> what are the issues you are trying to solve with this or what are the benefits?

@scoheb Description of commit and PR updated with reasons.

**Review by gabemontero — APPROVED** [review]

**Pull request merged**